graylog2-server/src/main/java/org/graylog2/rest/resources/system/RetentionStrategyResource.java

```diff
@@ -0,0 +1,154 @@
+        final IndexManagementConfig indexManagementConfig = clusterConfigService.get(IndexManagementConfig.class);
+        if (indexManagementConfig == null) {
+            throw new InternalServerErrorException("Couldn't retrieve index management configuration");
+        }
+
+        final String strategyName = indexManagementConfig.retentionStrategy();
+        final Provider<RetentionStrategy> provider = retentionStrategies.get(strategyName);
+        if (provider == null) {
+            throw new InternalServerErrorException("Couldn't retrieve retention strategy provider");
+        }
+
+        final RetentionStrategy retentionStrategy = provider.get();
+        @SuppressWarnings("unchecked")
+        final Class<RetentionStrategyConfig> configClass = (Class<RetentionStrategyConfig>) retentionStrategy.configurationClass();
+        final RetentionStrategyConfig config = clusterConfigService.get(configClass);
+
+        return RetentionStrategySummary.create(strategyName, config);
+    }
+
+    @PUT
+    @Path("config")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Timed
+    @ApiOperation(value = "Configuration of the current retention strategy",
+            notes = "This resource returns the configuration of the currently used retention strategy.")
+    public void config(@Valid @NotNull RetentionStrategySummary retentionStrategySummary) {
+        if (!retentionStrategies.containsKey(retentionStrategySummary.strategy())) {
+            throw new NotFoundException("Couldn't find retention strategy for given type " + retentionStrategySummary.strategy());
+        }
+
+        final IndexManagementConfig oldConfig = clusterConfigService.get(IndexManagementConfig.class);
+        if (oldConfig == null) {
+            throw new InternalServerErrorException("Couldn't retrieve index management configuration");
+        }
+
+        final IndexManagementConfig indexManagementConfig = IndexManagementConfig.create(
+                oldConfig.rotationStrategy(),
+                retentionStrategySummary.strategy()
+        );
+
+        clusterConfigService.write(retentionStrategySummary.config());
+        clusterConfigService.write(indexManagementConfig);
+    }
+
+    @GET
+    @Path("strategies")
+    @Timed
+    @ApiOperation(value = "List available retention strategies",
+            notes = "This resource returns a list of all available retention strategies on this Graylog node.")
+    public RetentionStrategies list() {
+        final Set<String> strategies = retentionStrategies.keySet();
+
+        return RetentionStrategies.create(strategies.size(), strategies);
+    }
+
+    @GET
+    @Path("strategies/{strategy}")
+    @Timed
+    @ApiOperation(value = "Show JSON schema for configuration of given retention strategies",
+            notes = "This resource returns a JSON schema for the configuration of the given retention strategy.")
+    public JsonSchema configSchema(@PathParam("strategy") @NotEmpty String strategyName) {
```

Comment:
Okay, cool. I just wanted to make sure that using json schema is a deliberate decision, that we want to continue to use it for other parts as well and that everyone knows about it. Otherwise it would become a "oh yeah, in this corner we are using json schema, but here we are using xyz". :wink: